### ISSUE TITLE
Overlapping storyboard commands behave differently from stable

### ISSUE THREAD

**Issue by MikiraSora** [ppy/osu#7257]
**Describe the bug:** 
In beatmap **[Sweet Dreams (11t dnb mix)](https://osu.ppy.sh/beatmapsets/499488)**
Playtime: **38330**ms and **95673**ms
part of osb code:
```
//line 309911 for part playtime 38330ms
Sprite,Foreground,CentreLeft,"sb\z37.png",32,234
 S,0,0,,0,0.5
 F,0,36015,36215,0,1
 MY,7,36015,36415,234,256
 R,0,36015,38415,0,0.128
 MX,0,36415,38015,32,31
 MY,0,36415,38015,256
 F,0,37387,37487,1,0
 F,0,37387,38415,0
 MY,6,38015,38415,256,271
 F,0,38215,38415,1,0
```

I am not sure if it's bug about osu!lazer storyboard implement , or storyboarder :(

**Screenshots or videos showing encountered issue:** 
lazer expect behavior screen:
![](https://puu.sh/EQFPT/e1b423bc94.png)

osu!stable:
![](https://puu.sh/EQFS6/d495007bb2.png)
**osu!lazer version:** 
[2019.1216.0](https://github.com/ppy/osu/releases/tag/2019.1216.0)

**Logs:**

**Comment by smoogipoo** [comment]
Will need to why stable handles this scenario this way, because it seems like lazer is doing what is expected by the storyboard (the last fade).

**Comment by MikiraSora** [comment]
Yep , so I think osu!stable/lazer could provide a standard .osb format checker to notice the storyboarders if they wrote bad/optimzable code.
Because I found some ranked beatmaps that storyboard code has bugs.

**Comment by smoogipoo** [comment]
Can you link to other beatmaps (and provide timestamps)? If this pattern is a common thing then we ought to fix it.

**Comment by MikiraSora** [comment]
If you mean diff behaviors between osu!lazer and osu!stable , currently i just know that two (more I forgot XD).
But I know some beatmap contains bugs/bad sytnax

**Comment by Wieku** [comment]
@MikiraSora by bad syntax and bugs don't you actually mean some shortcuts that make storyboards smaller in size?
Also - I don't see a problem in the `381480` beatmap you posted. It looks like some custom easing function from the generator Hollow Wings used.

**Comment by MikiraSora** [comment]
@Wieku Yep you are right , but this storyboard code contain unnesscery command like :
![image](https://user-images.githubusercontent.com/7549173/71231008-fbe46480-2326-11ea-9038-6deaa9e890fd.png)
it's about 10000+ commands .

**Comment by smoogipoo** [comment]
I've looked into this and it seems to be the difference between how transforms work in lazer vs stable.

Consider the case:

```
Transform1:     | -----------------------|
Transform2:                   |--------------------|
     Index:     0             1          2         3
```

In stable, T1  has an effect from t=0...2, and T2 has an effect from t=2...3.
In lazer, T1 has an effect from t=0...1, and T2 has an effect from t=1...3.

Lazer is doing this correctly so a framework fix isn't required, however slicing the transforms to match stable is non-trivial.

**Comment by peppy** [comment]
Please leave this open as it is still an issue.

**Comment by Shmiklak** [comment]
The storyboard does not look right on my recently [ranked beatmap](https://osu.ppy.sh/beatmapsets/2112582#osu/4521133) when playing in lazer. Specifically, on the attached screenshots we can see that for some reason lazer does not want to render the ECG, which I seems to be caused by this issue with overlapping commands. Below is the specific storyboard code that is used by the sprites on ECG animation:

```
Sprite,Background,Centre,"sb\glow.png",0,240
 L,39322,31
  F,0,0,353,0.7,0
  F,0,706,,0
 F,0,40028,,0
 S,0,40028,,0.1
 C,0,40028,,0,147,255
 MX,0,40028,,-107
 P,0,40028,62263,A
```

![image](https://github.com/ppy/osu/assets/20987315/73eab305-2d13-406f-9689-491b9ab50fbe)

lazer

![image](https://github.com/ppy/osu/assets/20987315/71dadc8e-14e1-4386-b52d-cdec029d5be9)

stable
